### ISSUE TITLE
Fast typing not working on second hand

### ISSUE THREAD

**Issue by techge** [rinx/trackpoint-iris#1]
Thanks for your guide, it helped me a lot figuring out the wiring for my Iris keyboard!

Unfortunately, I experience issues regarding the typing speed. The left hand (secondary hand, like in your pictures; =the hand without the trackpoint) is behaving as follows:

* if I am typing quite fast, some key strokes are not registered by the keyboard, especially if I am hitting a key on the other half this gets obvious, I get something like "try to type faste ras speed" were the "r" got registered only after I have typed the space already (I made quite sure, that it is not me, really ;) ).
* if I type one single key as often in row as I can, it becomes obvious, that a lot of strokes are missing on screen. Trying the same speed at the other half acts normal, everything is fine there
* connecting the USB cable on the left hand, disconnecting the TRRS-cable does not change a thing, so it seems like the trackpoint itself is not interfering but the settings are causing this issue somehow

Once I disable the trackpoint in firmware, everything is fine again - besides that I miss my beloved trackpoint feature, of course. I already changed the Micro Pro on left half, it was a lot of pain, would not recommend ;) so I am quite sure that it is the firmware.

Any one an idea, what is happening here?

**Comment by rinx** [comment]
@techge Hi. Thanks for sending the issue.

Unfortunately, my keyboard doesn't reproduce it.

By the way, do you use I2C for connecting right and left controllers over TRRS?
In my case, a serial connection is used. Not tested in I2C. So it may cause a problem like yours. :thinking:

**Comment by techge** [comment]
Sorry for the delay, but though annoying, it is not high on my TODO list :)

I think I found a solution and some hints for this issue. But it is weird to me that you do not experience this issue. Would you mind, hitting any key on each half very fast in an editor and try to find out if there are keys dropped on one of the halfs (presumably the one without a trackpoint)? Maybe you are just not having trouble with it for your type of ... typing :) but still having the same effect?

Anyway, I saw that [alonswartz mentioned some changes in the firmware](https://github.com/alonswartz/trackpoint#tmk-firmware-changes-and-tweaks) he made to regarding the "Trackpoint polling interval delay for dropped key strokes"... this sounds pretty much like my issue here so I tried the changes and... :tada: this actually helps. I am still not sure what exactly is happening, but it seems that this is a common issue related to polling for the trackpoint. I tested it the last minutes and for now it looks quite good. :smile:  quite happy about it...

**Comment by techge** [comment]
btw: using serial, not i2c and even tried the same config.h and rules.mk you have linked in this repo, did not help...

**Comment by rinx** [comment]
I've tried to type very fast in both sides, they completely work fine.

Because I didn't update the firmware for one and half a year, the base of the QMK firmware of my keyboards are old. I'm guessing something updated in the recently QMK may cause this problem.

**Comment by techge** [comment]
Just a small update for other users: My workarround now is to enable mouse support for right half (the trackpoint half) only, disabling it on the left part. Actually, I am not sure if this is the way I should have done it anyway...

The issue here is, that the left half was waiting for the trackpoint to initialize (but it is not present there...) which slowed down the response time, so by not enabling the feature there, everything is fine. Would be nice to have an automatic detection in qmk to avoid changing firmware settings for each half, but currently I am too lazy to request such feature.
